core/server/worker/src/main/java/alluxio/worker/DataServer.java

```diff
@@ -11,46 +11,15 @@
 
 package alluxio.worker;
 
-import alluxio.conf.ServerConfiguration;
-import alluxio.conf.PropertyKey;
-import alluxio.util.CommonUtils;
-
 import java.io.Closeable;
 import java.net.InetSocketAddress;
 import java.net.SocketAddress;
 
-import javax.annotation.concurrent.ThreadSafe;
-
 /**
  * Defines how to interact with a server running the data protocol.
  */
 public interface DataServer extends Closeable {
 
-  /**
-   * Factory for {@link DataServer}.
-   */
-  @ThreadSafe
-  class Factory {
-
-    private Factory() {} // prevent instantiation
-
-    /**
-     * Factory for {@link DataServer}.
-     *
-     * @param hostName the host name of the data server
-     * @param dataAddress the address of the data server
-     * @param worker the Alluxio worker handle
-     * @return the generated {@link DataServer}
-     */
-    public static DataServer create(final String hostName,
-        final SocketAddress dataAddress, final WorkerProcess worker) {
-      return CommonUtils.createNewClassInstance(
-          ServerConfiguration.<DataServer>getClass(PropertyKey.WORKER_DATA_SERVER_CLASS),
```

Comment:
there is only one implementation. I don't see why we want to classload DataServer

Reply:
IIRC there were more implementation pre-gRPC era. Now there is only `GrpcDataServer`.